spark/dl/src/main/scala/com/intel/analytics/bigdl/tensor/Tensor.scala

```diff
@@ -34,6 +34,18 @@ import scala.reflect.ClassTag
  * @tparam T should be Double or Float
  */
 trait Tensor[T] extends Serializable with TensorMath[T] with Activity {
+
+  /**
+   * @return whether this tensor is an empty tensor. Note that nDimension == 0 is not
+   *         sufficient to determine a tensor is empty, because a scalar tensor's nDimension
+   *         is also 0.
+   */
+  def isEmpty: Boolean
```

Comment:
can nElement == 0 check if a tensor is empty

Reply:
Yes, but it may iterate through all the dimensions if the tensor is not empty.